src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -534,6 +540,11 @@ def __init__(self):
         self._register_default_ciphers()
         self.activate_osrandom_engine()
 
+    def openssl_assert(self, ok):
+        if not ok:
+            errors = self._consume_errors()
+            raise UnhandledOpenSSLError("Unknown OpenSSL error", errors)
```

Comment:
Sounds good

On Fri, Sep 25, 2015 at 3:50 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/backend.py
> <https://github.com/pyca/cryptography/pull/2368#discussion_r40468961>:
>
> > @@ -534,6 +540,11 @@ def __init__(self):
> >          self._register_default_ciphers()
> >          self.activate_osrandom_engine()
> >
> > +    def openssl_assert(self, ok):
> > +        if not ok:
> > +            errors = self._consume_errors()
> > +            raise UnhandledOpenSSLError("Unknown OpenSSL error", errors)
>
> That seems reasonable. "Unknown OpenSSL error. Please file an issue at
> https://github.com/pyca/cryptography/issues with information on how to
> reproduce" sound good?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2368/files#r40468961>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084